test/metabase/models/user_test.clj

```diff
@@ -397,9 +397,9 @@
                (db/select-one-field :reset_token User :id user-id)))))
 
     (testing "should clear out all existing Sessions"
-      (mt/with-temp* [User    [{user-id :id}]
-                      Session [_ {:id (str (java.util.UUID/randomUUID)), :user_id user-id}]
-                      Session [_ {:id (str (java.util.UUID/randomUUID)), :user_id user-id}]]
```

Comment:
I made some tweaks to `with-temp` so it will error if `insert!` doesn't return a row (which seems to be the case for H2 for Tables without a single integer primary key column). It can't actually delete that row if it doesn't know what primary key to use for the row. So `with-temp` was actually busted in these cases.

In this case `insert!` is fine since the Session will get cascade deleted when the User gets deleted

Reply:
The issue is with H2/`clojure.java.jdbc` insert returning keys behavior... for H2 it doesn't return anything